flytekit/core/type_engine.py

```diff
@@ -22,6 +22,7 @@
 import msgpack
 from dataclasses_json import DataClassJsonMixin, dataclass_json
 from flyteidl.core import literals_pb2
+from fsspec.asyn import _run_coros_in_chunks  # pylint: disable=W0212
```

Comment:
Can we bring in this code from fsspec? I rather not depend on private API from another library.

Reply:
fsspec is not going anywhere. I don't understand why that particular method is private, but that function existed in its current form [since 2021](https://github.com/fsspec/filesystem_spec/blame/master/fsspec/asyn.py#L205-L233). IMO it's safe to assume that's going to live there for a long time.

Reply:
asked them to make it public: https://github.com/fsspec/filesystem_spec/issues/1740.  i think we should just use the private function until they make it public.  if they refuse, we can always copy later.  If we copy, i'll also be copying in the config code cuz I think it is nice to have some user facing controls around that.

Reply:
@thomasjpfan?  is this satisfactory?